### PR TITLE
Bump dependencies versions

### PR DESCRIPTION
**Pull request by grillorafael**
I'm trying to update the libraries on the project but I'm getting 403 on mesosphere maven.

```
error]         download error: Caught java.io.IOException: Server returned HTTP response code: 403 for URL: http://downloads.mesosphere.io/maven/com/twitter/util-core_2.11/6.37.1/util-core_2.11-6.37.1.pom (Server returned HTTP response code: 403 for URL: http://downloads.mesosphere.io/maven/com/twitter/util-core_2.11/6.37.1/util-core_2.11-6.37.1.pom) while downloading http://downloads.mesosphere.io/maven/com/twitter/util-core_2.11/6.37.1/util-core_2.11-6.37.1.pom
[error]         not found:
[error]             /Users/rafael/.ivy2/local/com.twitter/util-core_2.11/6.37.1/ivys/ivy.xml
[error]             https://repo1.maven.org/maven2/com/twitter/util-core_2.11/6.37.1/util-core_2.11-6.37.1.pom
[error]             https://jcenter.bintray.com/com/twitter/util-core_2.11/6.37.1/util-core_2.11-6.37.1.pom
[error]             http://repo.spray.io/com/twitter/util-core_2.11/6.37.1/util-core_2.11-6.37.1.pom
[error]             http://dl.bintray.com/emueller/maven/com/twitter/util-core_2.11/6.37.1/util-core_2.11-6.37.1.pom
[error]             http://repo.typesafe.com/typesafe/releases/com/twitter/util-core_2.11/6.37.1/util-core_2.11-6.37.1.pom
[error] Total time: 8 s, completed 28-Aug-2018 20:40:01
```

Any ideas?

# UPDATE:

1. This pull request updates all libraries (apart from `Twitter Common`)
2. Fix all warnings
3. Changes due to library updates
4. Refactor play controller to use `AbstractController` capabilities
    1. I would like some more thorough into those changes as I'm not sure all of them are sensible changes to be made

### CODE REVIEW

**Comment by kensipe** [comment]
checking...

**Comment by kensipe** [comment]
twitter repo isn't in our repo list... it will need to be added... I don't remember the details but I believe this is OSS and available.

**Comment by grillorafael** [comment]
ill Check.  Is there a cache in mesosphere maven ? Or they changed where it is hosted?

**Comment by grillorafael** [comment]
Weird that it seems to be looking at `_2.11`

**Comment by grillorafael** [comment]
tried updating only play version and got the same issue.

```
[error]         download error: Caught java.io.IOException: Server returned HTTP response code: 403 for URL: http://downloads.mesosphere.io/maven/com/typesafe/play/play-json_2.12/2.6.18/play-json_2.12-2.6.18.pom (Server returned HTTP response code: 403 for URL: http://downloads.mesosphere.io/maven/com/typesafe/play/play-json_2.12/2.6.18/play-json_2.12-2.6.18.pom) while downloading http://downloads.mesosphere.io/maven/com/typesafe/play/play-json_2.12/2.6.18/play-json_2.12-2.6.18.pom
[error]         not found:
[error]             /Users/rafael/.ivy2/local/com.typesafe.play/play-json_2.12/2.6.18/ivys/ivy.xml
[error]             https://repo1.maven.org/maven2/com/typesafe/play/play-json_2.12/2.6.18/play-json_2.12-2.6.18.pom
[error]             https://jcenter.bintray.com/com/typesafe/play/play-json_2.12/2.6.18/play-json_2.12-2.6.18.pom
[error]             http://repo.spray.io/com/typesafe/play/play-json_2.12/2.6.18/play-json_2.12-2.6.18.pom
[error]             http://dl.bintray.com/emueller/maven/com/typesafe/play/play-json_2.12/2.6.18/play-json_2.12-2.6.18.pom
[error]             http://repo.typesafe.com/typesafe/releases/com/typesafe/play/play-json_2.12/2.6.18/play-json_2.12-2.6.18.pom
[error] Total time: 12 s, completed 28-Aug-2018 21:29:06

```

maybe sbt is not dealing well with a 403 from mesosphere maven

**Comment by kensipe** [comment]
I'll take a look on this tomorrow US time if @alenkacz doesn't during EU time

**Comment by grillorafael** [comment]
Thanks @kensipe 

I'm on EU time at the moment. This PR is not yet finished and I created now to be able to get some help with this `403` issue. As soon as the dependencies can be downloaded properly I'll continue and start adjusting the code in case a dependency update requires it.

**Comment by grillorafael** [comment]
Hey guys, any updates on this issue?

**Comment by grillorafael** [comment]
Issue found.

Seems that `TwitterCommons` in the newer version required `finagle-core-java` which is not able to find and `play-json` don't have version `2.6.18` (only `2.6.10`).

I did those changes and I get dependencies to resolve.

**Comment by grillorafael** [comment]
Can someone label this pull request as work in progress since I don't have permissions to do it?

**Comment by grillorafael** [comment]
Updated the PR now

File: api/src/main/scala/dcos/metronome/api/ApiModule.scala [hunk 1]
```diff
@@ -16,46 +17,37 @@ import mesosphere.marathon.core.plugin.PluginManager
...
-import play.api.mvc.{ AnyContent, BodyParser }
+import play.api.mvc.ControllerComponents
 import play.api.routing.Router
 import router.Routes
 
 import scala.concurrent.ExecutionContext
 
-class ApiModule(
+class ApiModule(controllerComponents: ControllerComponents, assets: Assets, httpErrorHandler: HttpErrorHandler)(
+  implicit
```
Comment (ichernetsky) [inline api/src/main/scala/dcos/metronome/api/ApiModule.scala]: Why `ec` is made to be the very first argument, as it seems, everywhere?

File: api/src/main/scala/dcos/metronome/api/v1/controllers/JobScheduleController.scala [hunk 2]
```diff
@@ -7,22 +7,20 @@ import dcos.metronome.api.v1.models.schema._
 import dcos.metronome.jobspec.JobSpecService
 import dcos.metronome.model.{ JobId, JobSpec, ScheduleSpec }
 import mesosphere.marathon.plugin.auth._
-import JobId._
 import akka.stream.Materializer
 import mesosphere.marathon.metrics.Metrics
-import play.api.mvc.{ AnyContent, BodyParser, Result }
+import play.api.mvc._
```
Comment (ichernetsky) [inline api/src/main/scala/dcos/metronome/api/v1/controllers/JobScheduleController.scala]: Please avoid `_` imports.

File: api/src/main/scala/dcos/metronome/api/ApiModule.scala [hunk 3]
```diff
@@ -16,46 +17,37 @@ import mesosphere.marathon.core.plugin.PluginManager
...
+  ec:                 ExecutionContext,
```
Comment (kensipe) [inline api/src/main/scala/dcos/metronome/api/ApiModule.scala]: Why the change for `ec` from implicit?

File: api/src/main/scala/dcos/metronome/api/Authorization.scala [hunk 4]
```diff
@@ -51,41 +51,26 @@ case class AuthorizedRequest[Body](identity: Identity, request: Request[Body], a
...
+    Action andThen new AuthorizedActionBuilder()
+  def AuthorizedAction(identity: Identity): ActionBuilder[AuthorizedRequest, AnyContent] =
+    Action andThen new AuthorizedActionBuilder(Some(identity))
 
-  class AuthorizedActionBuilder(authorize: Option[Identity] = None) extends ActionBuilder[AuthorizedRequest, AnyContent] {
+  class AuthorizedActionBuilder(authorize: Option[Identity] = None) extends ActionRefiner[Request, AuthorizedRequest] {
+    override protected def executionContext: ExecutionContext = ec
 
-    def invokeBlock[A](request: Request[A], block: AuthorizedRequest[A] => Future[Result]) = {
+    override protected def refine[A](request: Request[A]): Future[Either[Result, AuthorizedRequest[A]]] = {
```
Comment (kensipe) [inline api/src/main/scala/dcos/metronome/api/Authorization.scala]: Is this all changes needed for updated version of `Play`?

File: api/src/main/scala/dcos/metronome/api/RestController.scala [hunk 5]
```diff
@@ -2,22 +2,22 @@ package dcos.metronome
 package api
 
 import com.eclipsesource.schema.SchemaValidator
-import com.wix.accord.{ Validator, Success, Failure }
+import com.wix.accord.{ Failure, Success, Validator }
```
Comment (kensipe) [inline api/src/main/scala/dcos/metronome/api/RestController.scala]: There are several changes here which are just reordering of imports... is there a reason?

File: api/src/main/scala/dcos/metronome/api/v0/controllers/ScheduledJobSpecController.scala [hunk 6]
```diff
@@ -12,18 +11,17 @@ import mesosphere.marathon.metrics.Metrics
...
-  jobSpecService:        JobSpecService,
-  val metrics:           Metrics,
-  val authenticator:     Authenticator,
-  val authorizer:        Authorizer,
-  val config:            ApiConfig,
-  val mat:               Materializer,
-  val defaultBodyParser: BodyParser[AnyContent])(implicit ec: ExecutionContext) extends Authorization {
+class ScheduledJobSpecController(cc: ControllerComponents)(
+  implicit
+  val ec: ExecutionContext, jobSpecService: JobSpecService, metrics: Metrics,
```
Comment (kensipe) [inline api/src/main/scala/dcos/metronome/api/v0/controllers/ScheduledJobSpecController.scala]: I prefer to either have all on 1 or 2 lines or all on separate lines... not a combo.   In this case.. all separate lines as it was previously.

File: api/src/main/scala/dcos/metronome/api/v1/controllers/JobRunController.scala [hunk 7]
```diff
@@ -9,20 +9,19 @@ import dcos.metronome.jobspec.JobSpecService
...
-  jobRunService:         JobRunService,
-  val metrics:           Metrics,
-  val authenticator:     Authenticator,
-  val authorizer:        Authorizer,
-  val config:            ApiConfig,
-  val mat:               Materializer,
-  val defaultBodyParser: BodyParser[AnyContent])(implicit ec: ExecutionContext) extends Authorization {
+class JobRunController(cc: ControllerComponents)(
+  implicit
+  ec: ExecutionContext, jobSpecService: JobSpecService,
```
Comment (kensipe) [inline api/src/main/scala/dcos/metronome/api/v1/controllers/JobRunController.scala]: again... 1 per line please

File: api/src/main/scala/dcos/metronome/api/v1/controllers/JobScheduleController.scala [hunk 8]
```diff
@@ -7,22 +7,20 @@ import dcos.metronome.api.v1.models.schema._
...
-  jobSpecService:        JobSpecService,
-  val metrics:           Metrics,
-  val authenticator:     Authenticator,
-  val authorizer:        Authorizer,
-  val config:            ApiConfig,
-  val mat:               Materializer,
-  val defaultBodyParser: BodyParser[AnyContent])(implicit ec: ExecutionContext) extends Authorization {
+class JobScheduleController(cc: ControllerComponents)(
+  implicit
+  ec: ExecutionContext, jobSpecService: JobSpecService, metrics: Metrics,
```
Comment (kensipe) [inline api/src/main/scala/dcos/metronome/api/v1/controllers/JobScheduleController.scala]: ditto

File: api/src/main/scala/dcos/metronome/api/v1/controllers/JobSpecController.scala [hunk 9]
```diff
@@ -10,24 +10,24 @@ import dcos.metronome.jobrun.JobRunService
 import dcos.metronome.jobspec.JobSpecService
 import dcos.metronome.model.{ JobId, JobSpec }
 import mesosphere.marathon.plugin.auth._
-import JobId._
 import akka.stream.Materializer
 import mesosphere.marathon.metrics.Metrics
-import play.api.mvc.{ AnyContent, BodyParser, Result }
+import play.api.mvc._
```
Comment (kensipe) [inline api/src/main/scala/dcos/metronome/api/v1/controllers/JobSpecController.scala]: no `_` for imports

File: api/src/main/scala/dcos/metronome/api/ApiModule.scala (hunk 1 above)
Comment (grillorafael) [reply api/src/main/scala/dcos/metronome/api/ApiModule.scala]: No reason in particular. Ve was already implicit so I just added the other ones after. 

I can change it but my question as well is if we should keep everything as implicit or not. I made the change but I’m not sure if it is any better

**Comment by ichernetsky** [comment]
BTW, the first green build.

File: api/src/main/scala/dcos/metronome/api/v1/controllers/JobSpecController.scala [hunk 10]
```diff
@@ -69,23 +69,22 @@ class JobSpecController(
...
-              case _: JobSpecDoesNotExist => NotFound(UnknownJob(id))
+      def deleteJobSpec(jobSpec: JobSpec): Future[Result] = {
+        jobRunService.activeRuns(id).flatMap {
+          case _ :: _ if !stopCurrentJobRuns =>
+            Future.successful(Conflict(ErrorDetail("There are active job runs. Override with stopCurrentJobRuns=true")))
+          case runs =>
+            val f = for {
+              _ <- Future.sequence(runs.map(run => jobRunService.killJobRun(run.jobRun.id)))
+              deleteRes <- jobSpecService.deleteJobSpec(id)
+            } yield Ok(deleteRes)
```
Comment (kensipe) [inline api/src/main/scala/dcos/metronome/api/v1/controllers/JobSpecController.scala]: impressive... this terse and we need to make it as readable as possible.... can we rename or fully name `deleteRes`

File: api/src/main/scala/dcos/metronome/api/v1/controllers/LaunchQueueController.scala [hunk 11]
```diff
@@ -7,16 +7,17 @@ import dcos.metronome.api.{ ApiConfig, Authorization }
...
-  val authenticator:     Authenticator,
-  val authorizer:        Authorizer,
-  val config:            ApiConfig,
-  val mat:               Materializer,
-  val defaultBodyParser: BodyParser[AnyContent]) extends Authorization {
+import scala.concurrent.ExecutionContext
+
+class LaunchQueueController(cc: ControllerComponents)(
+  implicit
+  ec: ExecutionContext, queueService: LaunchQueueService, metrics: Metrics,
```
Comment (kensipe) [inline api/src/main/scala/dcos/metronome/api/v1/controllers/LaunchQueueController.scala]: ditto

File: api/src/main/scala/dcos/metronome/api/ApiModule.scala (hunk 1 above)
Comment (ichernetsky) [reply api/src/main/scala/dcos/metronome/api/ApiModule.scala]: I don't see a reason to change it. + the diff would be smaller (less reformatting)

File: api/src/main/scala/dcos/metronome/api/RestController.scala (hunk 5 above)
Comment (ichernetsky) [reply api/src/main/scala/dcos/metronome/api/RestController.scala]: It is IDEA. +1 to the alphabetical ordering

File: api/src/main/scala/dcos/metronome/api/ApiModule.scala (hunk 1 above)
Comment (grillorafael) [reply api/src/main/scala/dcos/metronome/api/ApiModule.scala]: I’ll rollback this update the

File: api/src/main/scala/dcos/metronome/api/Authorization.scala (hunk 4 above)
Comment (grillorafael) [reply api/src/main/scala/dcos/metronome/api/Authorization.scala]: This wasn’t necessary but getting rid of some of the warns it was easier to use ActionRefiner rather than the previous one. (It doesn’t require you to specify a bodyparser for example)

File: api/src/main/scala/dcos/metronome/api/RestController.scala (hunk 5 above)
Comment (grillorafael) [reply api/src/main/scala/dcos/metronome/api/RestController.scala]: Yes. Just IDEA reordering it

File: api/src/main/scala/dcos/metronome/api/v1/controllers/JobSpecController.scala (hunk 10 above)
Comment (grillorafael) [reply api/src/main/scala/dcos/metronome/api/v1/controllers/JobSpecController.scala]: Yes. Just to give context, I was removing a warn because of the two awaits that were not going anywhere (just generated side effect or an exception). Tried to refactor in a way to make the later step be a for comprehension noting that they had to run in order and either could fail. 

But yes will rename it

**Comment by grillorafael** [comment]
Late in EU now, I’ll work on the comments sometime tomorrow but thanks for the thorough look on it! Few free to send all comments now and I’ll send an update tomorrow

File: api/src/main/scala/dcos/metronome/api/ApiModule.scala (hunk 1 above)
Comment (grillorafael) [reply api/src/main/scala/dcos/metronome/api/ApiModule.scala]: Actually, I think read your comment wrong. You prefer keeping like this or revert and not have ec as the first one?

File: api/src/main/scala/dcos/metronome/api/RestController.scala (hunk 5 above)
Comment (grillorafael) [reply api/src/main/scala/dcos/metronome/api/RestController.scala]: I'll mark as resolve and keep the change. Feel free to comment again if that is not the desired result.

File: api/src/main/scala/dcos/metronome/api/v1/controllers/JobSpecController.scala (hunk 10 above)
Comment (grillorafael) [reply api/src/main/scala/dcos/metronome/api/v1/controllers/JobSpecController.scala]: renamed to `deletedJobSpec`

File: api/src/main/scala/dcos/metronome/api/ApiModule.scala (hunk 3 above)
Comment (grillorafael) [reply api/src/main/scala/dcos/metronome/api/ApiModule.scala]: `ec` was already previously `implicit` (Updated due to deprecated play ec import)

Comment (grillorafael) [reply api/src/main/scala/dcos/metronome/api/ApiModule.scala]: I'll mark this as resolved as well

**Comment by grillorafael** [comment]
There are 2 unresolved comments here. Please comment in case I need to do any more changes :mega:

**Comment by grillorafael** [comment]
I don't think there is anything else left. I left https://github.com/dcos/metronome/pull/254#discussion_r215053229 and https://github.com/dcos/metronome/pull/254#discussion_r215051232 as unresolved comments because I didn't know if you guys were happy with them. But if you guys are happy then I won't add anything to this PR anymore

**Comment by kensipe** [comment]
I'm pretty happy... I triggered the build... I'll update the label.

**Comment by grillorafael** [comment]
I ran all the tests locally now. Seems like everything passed. Can you send me the message or the log of why the build failed?

**Comment by kensipe** [comment]
@grillorafael looks like an env issue... looking to resolve it and re-run

File: api/src/main/scala/dcos/metronome/api/ApiModule.scala (hunk 1 above)
Comment (ichernetsky) [reply api/src/main/scala/dcos/metronome/api/ApiModule.scala]: As far as I know, by putting `implicit ec: ...` first, the rest of parameters are made implicit too. Having as few implicts as possible makes the code more readable. What do you think?

Comment (ichernetsky) [reply api/src/main/scala/dcos/metronome/api/ApiModule.scala]: Meaning, `ec` should be placed as it is on master.

**Comment by kensipe** [comment]
@grillorafael this generally looks good... however we are looking to have as few implicit as possible.   It looks like you have extended the list.   Could we switch that back?  It isn't part of the need for the bump in dependencies as well.     That change and we can get this merged.

**Comment by grillorafael** [comment]
Will do the change 👍

File: api/src/main/scala/dcos/metronome/api/ApiModule.scala (hunk 1 above)
Comment (timcharper) [reply api/src/main/scala/dcos/metronome/api/ApiModule.scala]: Agreed, we're conservative with implicits round these parts :)

File: api/src/main/scala/dcos/metronome/api/v1/controllers/JobSpecController.scala [hunk 12]
```diff
@@ -10,24 +10,24 @@ import dcos.metronome.jobrun.JobRunService
...
-  jobRunService:         JobRunService,
-  jobInfoService:        JobInfoService,
-  val metrics:           Metrics,
-  val authenticator:     Authenticator,
-  val authorizer:        Authorizer,
-  val config:            ApiConfig,
-  val mat:               Materializer,
-  val defaultBodyParser: BodyParser[AnyContent])(implicit ec: ExecutionContext) extends Authorization {
+class JobSpecController(cc: ControllerComponents)(
+  implicit
```
Comment (timcharper) [inline api/src/main/scala/dcos/metronome/api/v1/controllers/JobSpecController.scala]: Let's please be explicit here.

File: api/src/main/scala/dcos/metronome/api/v1/controllers/JobRunController.scala [hunk 13]
```diff
@@ -9,20 +9,21 @@ import dcos.metronome.jobspec.JobSpecService
...
-  jobSpecService:        JobSpecService,
-  jobRunService:         JobRunService,
-  val metrics:           Metrics,
-  val authenticator:     Authenticator,
-  val authorizer:        Authorizer,
-  val config:            ApiConfig,
-  val mat:               Materializer,
-  val defaultBodyParser: BodyParser[AnyContent])(implicit ec: ExecutionContext) extends Authorization {
+class JobRunController(cc: ControllerComponents)(
+  implicit
```
Comment (timcharper) [inline api/src/main/scala/dcos/metronome/api/v1/controllers/JobRunController.scala]: explicit desired

File: api/src/main/scala/dcos/metronome/api/v0/controllers/ScheduledJobSpecController.scala [hunk 14]
```diff
@@ -12,18 +11,19 @@ import mesosphere.marathon.metrics.Metrics
...
-class ScheduledJobSpecController(
-  jobSpecService:        JobSpecService,
-  val metrics:           Metrics,
-  val authenticator:     Authenticator,
-  val authorizer:        Authorizer,
-  val config:            ApiConfig,
-  val mat:               Materializer,
-  val defaultBodyParser: BodyParser[AnyContent])(implicit ec: ExecutionContext) extends Authorization {
+class ScheduledJobSpecController(cc: ControllerComponents)(
+  implicit
```
Comment (timcharper) [inline api/src/main/scala/dcos/metronome/api/v0/controllers/ScheduledJobSpecController.scala]: explicit desired

File: api/src/main/scala/dcos/metronome/api/v1/controllers/LaunchQueueController.scala [hunk 15]
```diff
@@ -7,16 +7,19 @@ import dcos.metronome.api.{ ApiConfig, Authorization }
...
-  val metrics:           Metrics,
-  val authenticator:     Authenticator,
-  val authorizer:        Authorizer,
-  val config:            ApiConfig,
-  val mat:               Materializer,
-  val defaultBodyParser: BodyParser[AnyContent]) extends Authorization {
+import scala.concurrent.ExecutionContext
+
+class LaunchQueueController(cc: ControllerComponents)(
+  implicit
```
Comment (timcharper) [inline api/src/main/scala/dcos/metronome/api/v1/controllers/LaunchQueueController.scala]: explicit desired

File: api/src/test/scala/dcos/metronome/api/WithApplicationComponents.scala [hunk 16]
```diff
@@ -102,30 +102,23 @@ class MockApiComponents(context: Context) extends BuiltInComponentsFromContext(c
 
   override lazy val httpErrorHandler = new ErrorHandler
 
-  lazy val actorsModule = new ActorsModule(actorSystem)
+  implicit lazy val actorsModule: ActorsModule = new ActorsModule(actorSystem)
```
Comment (timcharper) [inline api/src/test/scala/dcos/metronome/api/WithApplicationComponents.scala]: explicit desired

**Comment by timcharper** [comment]
Thanks @grillorafael !

**Comment by timcharper** [comment]
Realizing just now that it's not really clear since we don't have lots of (any?) contributing guidelines for Metronome :) What we have written for Marathon applies, though:

https://github.com/mesosphere/marathon/blob/master/docs/docs/code-culture.md#on-implicits :)

File: jobs/src/main/scala/dcos/metronome/model/CronSpec.scala [hunk 17]
```diff
@@ -20,12 +20,12 @@ class CronSpec(val cron: Cron) {
 
   def nextExecution(from: ZonedDateTime): ZonedDateTime = {
     val fromDateTime: ThreeTenZonedDateTime = javaTimeToThreetenTime(from)
-    threetenToJavaTime(executionTime.nextExecution(fromDateTime).get())
+    executionTime.nextExecution(threetenToJavaTime(fromDateTime)).get()
```
Comment (grillorafael) [inline jobs/src/main/scala/dcos/metronome/model/CronSpec.scala]: Can someone have a special look at this change as well?

It didn't compile for me what we have in master.

Comment (grillorafael) [reply jobs/src/main/scala/dcos/metronome/model/CronSpec.scala]: @kensipe

**Comment by grillorafael** [comment]
PR updated. I brought back `wire[Controller]` to Api module. If you prefer explicit parameter passing in there let me know and I'll rollback. I added the changes to my latest commit https://github.com/dcos/metronome/pull/254/commits/14a1dafcfc1956ac93ffba2f8685d06bf7ae26a1

File: jobs/src/main/scala/dcos/metronome/model/CronSpec.scala (hunk 17 above)
Comment (kensipe) [reply jobs/src/main/scala/dcos/metronome/model/CronSpec.scala]: master didn't compile?   that seems strange.. it compiles locally and on jenkins.    just tested again... you sure?

Comment (kensipe) [reply jobs/src/main/scala/dcos/metronome/model/CronSpec.scala]: if you switched while using IDEA and didn't refresh the sbt... this would have happened... could you check again?

Comment (kensipe) [reply jobs/src/main/scala/dcos/metronome/model/CronSpec.scala]: regarding the change... cron-utils 7.0.0 must have move away from 310 and strictly to Java dates... I will take a look after this PR.  We may not need the conversations anymore.   The frustration is that they have had 9 releases without release notes with 6.0.3 being the last documented release 👎 .

Comment (grillorafael) [reply jobs/src/main/scala/dcos/metronome/model/CronSpec.scala]: Yes I believe I missed an sbt reload.

Comment (grillorafael) [reply jobs/src/main/scala/dcos/metronome/model/CronSpec.scala]: Yes I believe I missed an sbt reload.
